tests/test_scripts.py

```diff
@@ -84,13 +88,42 @@ def test_proto_conversion(tmpdir):
     assert p.stdout.startswith(b'{\n  "meta": ') or p.stdout.startswith(b'{\r\n  "meta": ')
 
 
-def test_detect_duplicate_features():
-    new_rule_path = "collection/credit-card/parse-credit-card-information.yml"
-    args = [
-        get_rules_path(),
-        os.path.join(get_rules_path(), new_rule_path),
-    ]
-    expected_overlaps = 49
+def run_detect_duplicate_features(rule_path):
+    # rule_path = "collection/credit-card/parse-credit-card-information.yml"
+    args = [get_rules_path(), rule_path]
     script_path = get_script_path("detect_duplicate_features.py")
-    p = run_program(script_path, args)
-    assert p.returncode == expected_overlaps
+    args = [sys.executable] + [script_path] + args
+    print(f"running: '{args}'")
+    return subprocess.run(args)
+
+
+def test_detect_duplicate_features(z9324d_extractor, tmpdir):
+    # tests a single rule can be loaded successfully
+    RULE_CONTENT = textwrap.dedent(
+        """
+        rule:
+          meta:
+            name: Test Rule
+            scope: function
+          features:
+            - string: "sites.ini"
+        """
+    )
+    expected_overlaps = 3
```

Comment:
what are the 3 features here? I only see one feature in the rule.

Reply:
```
rules\collection\file-managers\gather-staff-ftp-information.yml
rules\collection\file-managers\gather-leapftp-information.yml
rules\collection\file-managers\gather-3d-ftp-information.yml
```
These 3 rules have the same feature` - string: "sites.ini"`. Expected Overlaps are the number of rules which have overlapping features as test rule.

Reply:
Ah yeah. Again, let's create a hard-coded rule set so that changes to the real set don't mess with our tests (like when we'd add a new rule that uses this feature and suddenly tests fail).

Reply:
I made a mistake by not considering the potential issue of adding a new rule that could cause the tests to fail. As you mentioned it would be a good idea to create a new RuleSet and test the script within it for each rule.